packages/hdwallet-provider/src/constructor/getOptions.ts

```diff
@@ -20,29 +17,31 @@ const isPrivateKeys = (
 const isPrivateKey = (
   credentials: LegacyConstructor.Credentials
 ): credentials is PrivateKey =>
-  !isPrivateKeys(credentials) && !isMnemonicPhrase(credentials);
+  typeof credentials === "string" &&
+  credentials.length === 64 &&
+  // this is added since parseInt(mnemonic) should equal NaN and private keys
+  // should parse into a valid number
+  parseInt(credentials) !== NaN;
 
 // turn polymorphic first argument into { mnemonic } or { privateKeys }
 const getSigningAuthorityOptions = (
   credentials: LegacyConstructor.Credentials
 ): Constructor.SigningAuthority => {
-  if (isMnemonicPhrase(credentials)) {
+  if (isPrivateKey(credentials)) {
     return {
-      mnemonic: {
-        phrase: credentials
-      }
+      privateKeys: [credentials]
     };
   } else if (isPrivateKeys(credentials)) {
     return {
       privateKeys: credentials
     };
-  } else if (isPrivateKey(credentials)) {
-    // if(...) included for explicitness
+  } else if (isMnemonicPhrase(credentials)) {
```

Comment:
Per my comments above, I think the function `isMnemonicPhrase` should be eliminated if it's not going to actually perform any validation, and instead this condition should just be `typeof credentials === "string"`; you could then add a comment noting that in this case you assume it's a mnemonic.

Reply:
Makes sense!